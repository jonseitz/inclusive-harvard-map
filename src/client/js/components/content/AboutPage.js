import React from 'react';
import ReactMarkdown from 'react-markdown';
import { Typography } from '@material-ui/core';

/**
 * Renders a simple description of the app
 * @extends React.Component
 */

const content = `
Our goal is to help everyone feel welcome on Harvard's campus by providing a discreet and efficient way to find gender-neutral restrooms and lactation rooms on campus.

This app was build by Jonathan Seitz, Application Developer in the Computing Department at the Harvard John A. Paulson School of Engineering and Applied Sciences, as a Capstone Project for the Design Media Design program at the Harvard Extension School. It is currently a pilot, so some data may be inaccurate or incomplete.

Building floorplans were provided by the Harvard Planning Office. 

Mapping data is provided by OpenStreetMaps.
`;

class AboutPage extends React.Component {
  render() {
    return (
      <Typography variant="body2">
        <ReactMarkdown>
          {content}
        </ReactMarkdown>
      </Typography>);
  }
}

export default AboutPage;
