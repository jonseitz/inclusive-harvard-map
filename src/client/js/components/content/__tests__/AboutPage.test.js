import React from 'react';
import enzyme from 'enzyme';
import AboutPage from '../AboutPage';

describe('AboutPage', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = enzyme.shallow(<AboutPage />);
  });
  it('Should render the content without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
