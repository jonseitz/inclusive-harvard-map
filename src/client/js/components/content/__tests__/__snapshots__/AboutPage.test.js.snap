// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage Should render the content without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      >
        
Our goal is to help everyone feel welcome on Harvard's campus by providing a discreet and efficient way to find gender-neutral restrooms and lactation rooms on campus.

This app was build by Jonathan Seitz, Application Developer in the Computing Department at the Harvard John A. Paulson School of Engineering and Applied Sciences, as a Capstone Project for the Design Media Design program at the Harvard Extension School. It is currently a pilot, so some data may be inaccurate or incomplete.

Building floorplans were provided by the Harvard Planning Office. 

Mapping data is provided by OpenStreetMaps.

      </ReactMarkdown>,
      "variant": "body2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "astPlugins": Array [],
        "children": "
Our goal is to help everyone feel welcome on Harvard's campus by providing a discreet and efficient way to find gender-neutral restrooms and lactation rooms on campus.

This app was build by Jonathan Seitz, Application Developer in the Computing Department at the Harvard John A. Paulson School of Engineering and Applied Sciences, as a Capstone Project for the Design Media Design program at the Harvard Extension School. It is currently a pilot, so some data may be inaccurate or incomplete.

Building floorplans were provided by the Harvard Planning Office. 

Mapping data is provided by OpenStreetMaps.
",
        "escapeHtml": true,
        "plugins": Array [],
        "rawSourcePos": false,
        "renderers": Object {},
        "skipHtml": false,
        "sourcePos": false,
        "transformLinkUri": [Function],
      },
      "ref": null,
      "rendered": "
Our goal is to help everyone feel welcome on Harvard's campus by providing a discreet and efficient way to find gender-neutral restrooms and lactation rooms on campus.

This app was build by Jonathan Seitz, Application Developer in the Computing Department at the Harvard John A. Paulson School of Engineering and Applied Sciences, as a Capstone Project for the Design Media Design program at the Harvard Extension School. It is currently a pilot, so some data may be inaccurate or incomplete.

Building floorplans were provided by the Harvard Planning Office. 

Mapping data is provided by OpenStreetMaps.
",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          sourcePos={false}
          transformLinkUri={[Function]}
        >
          
Our goal is to help everyone feel welcome on Harvard's campus by providing a discreet and efficient way to find gender-neutral restrooms and lactation rooms on campus.

This app was build by Jonathan Seitz, Application Developer in the Computing Department at the Harvard John A. Paulson School of Engineering and Applied Sciences, as a Capstone Project for the Design Media Design program at the Harvard Extension School. It is currently a pilot, so some data may be inaccurate or incomplete.

Building floorplans were provided by the Harvard Planning Office. 

Mapping data is provided by OpenStreetMaps.

        </ReactMarkdown>,
        "variant": "body2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "astPlugins": Array [],
          "children": "
Our goal is to help everyone feel welcome on Harvard's campus by providing a discreet and efficient way to find gender-neutral restrooms and lactation rooms on campus.

This app was build by Jonathan Seitz, Application Developer in the Computing Department at the Harvard John A. Paulson School of Engineering and Applied Sciences, as a Capstone Project for the Design Media Design program at the Harvard Extension School. It is currently a pilot, so some data may be inaccurate or incomplete.

Building floorplans were provided by the Harvard Planning Office. 

Mapping data is provided by OpenStreetMaps.
",
          "escapeHtml": true,
          "plugins": Array [],
          "rawSourcePos": false,
          "renderers": Object {},
          "skipHtml": false,
          "sourcePos": false,
          "transformLinkUri": [Function],
        },
        "ref": null,
        "rendered": "
Our goal is to help everyone feel welcome on Harvard's campus by providing a discreet and efficient way to find gender-neutral restrooms and lactation rooms on campus.

This app was build by Jonathan Seitz, Application Developer in the Computing Department at the Harvard John A. Paulson School of Engineering and Applied Sciences, as a Capstone Project for the Design Media Design program at the Harvard Extension School. It is currently a pilot, so some data may be inaccurate or incomplete.

Building floorplans were provided by the Harvard Planning Office. 

Mapping data is provided by OpenStreetMaps.
",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
